packages/wrangler/src/pipelines/index.ts

```diff
@@ -0,0 +1,459 @@
+						);
+					}
+				}
+
+				if (args.filepath) {
+					pipelineConfig.destination.path.filepath = args.filepath;
+				}
+				if (args.filename) {
+					pipelineConfig.destination.path.filename = args.filename;
+				}
+
+				logger.log(`🌀 Creating pipeline named "${name}"`);
+				const pipeline = await createPipeline(accountId, pipelineConfig);
+				await metrics.sendMetricsEvent("create pipeline", {
+					sendMetrics: config.send_metrics,
+				});
+
+				logger.log(
+					`✅ Successfully created pipeline "${pipeline.name}" with ID ${pipeline.id}`
+				);
+				logger.log("🎉 You can now send data to your pipeline!");
+				logger.log(
+					`Example: curl "${pipeline.endpoint}" -d '[{"foo": "bar"}]'`
+				);
+			}
+		)
+		.command(
+			"list",
+			"List current pipelines",
+			(yargs) => yargs,
+			async (args) => {
+				const config = readConfig(args.config, args);
+				const accountId = await requireAuth(config);
+
+				// TODO: we should show bindings & transforms if they exist for given ids
+				const pipelines = await listPipelines(accountId);
+				await metrics.sendMetricsEvent("list pipelines", {
+					sendMetrics: config.send_metrics,
+				});
+
+				logger.table(
+					pipelines.map((pipeline) => ({
+						name: pipeline.name,
+						id: pipeline.id,
+						endpoint: pipeline.endpoint,
+					}))
+				);
+			}
+		)
+		.command(
+			"show <pipeline>",
+			"Show a pipeline configuration",
+			(yargs) => {
+				return yargs.positional("pipeline", {
+					type: "string",
+					describe: "The name of the pipeline to show",
+					demandOption: true,
+				});
+			},
+			async (args) => {
+				await printWranglerBanner();
+				const config = readConfig(args.config, args);
+				const accountId = await requireAuth(config);
+				const name = args.pipeline;
+
+				if (!name.match(/^[a-zA-Z0-9-]+$/)) {
+					throw new Error("must provide a valid pipeline name");
+				}
+
+				logger.log(`Retrieving config for pipeline "${name}".`);
+				const pipeline = await getPipeline(accountId, name);
+				await metrics.sendMetricsEvent("show pipeline", {
+					sendMetrics: config.send_metrics,
+				});
+
+				logger.log(JSON.stringify(pipeline, null, 2));
+			}
+		)
+		.command(
+			"update <pipeline>",
+			"Update a pipeline",
+			(yargs) => {
+				return addCreateAndUpdateOptions(yargs)
+					.positional("pipeline", {
+						describe: "The name of the pipeline to update",
+						type: "string",
+						demandOption: true,
+					})
+					.option("r2", {
+						type: "string",
+						describe: "Destination R2 bucket name",
+						demandOption: false,
+					});
+			},
+			async (args) => {
+				await printWranglerBanner();
+
+				const name = args.pipeline;
+				// only the fields set will be updated - other fields will use the existing config
+				const config = readConfig(args.config, args);
+				const accountId = await requireAuth(config);
+
+				const pipelineConfig = await getPipeline(accountId, name);
```

Comment:
What should happen if this pipeline doesn't exist?

Reply:
Exception gets thrown that shows the error message that the pipeline doesn't exist:

```
Pipeline does not exist [code: 1000]
```